api/ocm/extensions/repositories/genericocireg/componentversion.go

```diff
@@ -304,15 +325,49 @@ func (c *ComponentVersionContainer) AddBlob(blob cpi.BlobAccess, refName string,
 		return nil, errors.New("a resource has to be defined")
 	}
 
+	size := blob.Size()
+	var refs []string
+	if c.comp.repo.blobLimit > 0 && size != blobaccess.BLOB_UNKNOWN_SIZE && size > c.comp.repo.blobLimit {
```

Comment:
Hm, since you have warding code in chunk for concurrency, I'm guessing this can be called concurrently. Meaning `blobLimit` here, could potentially be changed mid execution.

I suggest putting it outside as a variable like `size` and use that instead of keep calling it.

Reply:
you are right, fixed.